sycamore/transforms/embed.py

```diff
@@ -85,7 +92,69 @@ def generate_embeddings(self, doc_batch: list[Document]) -> list[Document]:
         text_batch = ["" if doc.text_representation is None else doc.text_representation for doc in doc_batch]
         embeddings = self._transformer.encode(text_batch, batch_size=self.model_batch_size, device=self.device)
         for doc, embedding in zip(doc_batch, embeddings):
-            doc.embedding = embedding
+            doc.embedding = embedding.tolist()
+
+        return doc_batch
+
+
+class OpenAIEmbeddingModels(Enum):
+    TEXT_EMBEDDING_ADA_002 = "text-embedding-ada-002"
+
+
+class OpenAIEmbedder(Embedder):
+    """Embedder implementation using the OpenAI embedding API.
+
+    Args:
+        model_name: The name of the OpenAI embedding model to use.
+        batch_size: The Ray batch size.
+        model_batch_size: The number of documents to send in a single OpenAI request.
+    """
+
+    def __init__(
+        self,
+        model_name: str = OpenAIEmbeddingModels.TEXT_EMBEDDING_ADA_002.value,
+        batch_size: Optional[int] = None,
+        model_batch_size: int = 100,
+        api_key: Optional[str] = None,
```

Comment:
could we explicitly set device it uses, otherwise, it might set to cuda if system has gpu available, or you could change the logic for setting device in base class.

Reply:
Doesn't the super call on line 120 explicitly set it to CPU? That was my intention anyway.